source-test/File.ts

```diff
@@ -1,661 +1,1617 @@
+  describe('inspect()', function() {
+
+    it('returns correct format when no contents and no path', function(done) {
+      const file = new Vinyl();
+      expect(file.inspect()).toEqual('<File >');
       done();
     });
 
-    it('should return correct format when Buffer and no path', done => {
+    it('returns correct format when Buffer contents and no path', function(done) {
       const val = new Buffer('test');
-      const file = new File({
-        contents: val
-      });
-      file.inspect().should.equal('<File <Buffer 74 65 73 74>>');
+      const file = new Vinyl({ contents: val });
+      expect(file.inspect()).toEqual('<File <Buffer 74 65 73 74>>');
       done();
     });
 
-    it('should return correct format when Buffer and relative path', done => {
+    it('returns correct format when Buffer contents and relative path', function(done) {
       const val = new Buffer('test');
-      const file = new File({
+      const file = new Vinyl({
         cwd: '/',
         base: '/test/',
         path: '/test/test.coffee',
-        contents: val
+        contents: val,
       });
-      file.inspect().should.equal('<File "test.coffee" <Buffer 74 65 73 74>>');
+      expect(file.inspect()).toEqual('<File "test.coffee" <Buffer 74 65 73 74>>');
       done();
     });
 
-    it('should return correct format when Buffer and only path and no base', done => {
-      const val = new Buffer('test');
-      const file = new File({
+    it('returns correct format when Stream contents and relative path', function(done) {
+      const file = new Vinyl({
         cwd: '/',
+        base: '/test/',
         path: '/test/test.coffee',
-        contents: val
+        contents: from([]),
       });
-      delete file.basename;
-      file.inspect().should.equal('<File "/test/test.coffee" <Buffer 74 65 73 74>>');
+      expect(file.inspect()).toEqual('<File "test.coffee" <CloneableStream>>');
       done();
     });
 
-    it('should return correct format when Stream and relative path', done => {
-      const file = new File({
+    it('returns correct format when null contents and relative path', function(done) {
+      const file = new Vinyl({
         cwd: '/',
         base: '/test/',
         path: '/test/test.coffee',
-        contents: new Stream.PassThrough()
+        contents: null,
       });
-      file.inspect().should.equal('<File "test.coffee" <PassThroughStream>>');
+      expect(file.inspect()).toEqual('<File "test.coffee">');
       done();
     });
+  });
 
-    it('should return correct format when null and relative path', done => {
-      const file = new File({
-        cwd: '/',
-        base: '/test/',
-        path: '/test/test.coffee',
-        contents: null
-      });
-      file.inspect().should.equal('<File "test.coffee">');
+  describe('contents get/set', function() {
+
+    it('returns _contents', function(done) {
+      const val = new Buffer('test');
+      const file: Vinyl & {_contents?: Buffer} = new Vinyl();
```

Comment:
👍 This is what I'm looking for - showing how to accomplish stuff without casting